R/type-vctr.R

```diff
@@ -115,25 +115,23 @@ vec_ptype2.vctrs_vctr <- function(x, y, ..., x_arg = "", y_arg = "") {
 
 #' @method vec_cast vctrs_vctr
 #' @export
-vec_cast.vctrs_vctr <- function(x, to, ...) UseMethod("vec_cast.vctrs_vctr")
+vec_cast.vctrs_vctr <- function(x, to, ...) {
+  UseMethod("vec_cast.vctrs_vctr")
+}
 
-#' @method vec_cast.vctrs_vctr default
-#' @export
-vec_cast.vctrs_vctr.default <- function(x, to, ...) {
+vctr_cast <- function(x, to, ..., x_arg = "", to_arg = "") {
   # These are not strictly necessary, but make bootstrapping a new class
   # a bit simpler
   if (is.object(x)) {
-    attr_x <- utils::modifyList(attributes(x), list(names = NULL))
-    attr_y <- utils::modifyList(attributes(to), list(names = NULL))
-
-    if (identical(attr_x, attr_y)) {
-      return(x)
+    if (is_same_type(x, to)) {
```

Comment:
This is a behaviour change, right? Is it worth a news bullet?

Reply:
This is a slight change, I don't think it's worth a bullet since this is rather esoteric.

Reply:
Isn't it the root of a bunch of the "can't cast <x> to <x>" issues?

Reply:
I don't think so, these have been fixed already.